src/components/MusicMarkdownNavbar.js

```diff
@@ -2,59 +2,57 @@ import Navbar from 'react-bootstrap/Navbar';
 import NavDropdown from 'react-bootstrap/NavDropdown';
 import Nav from 'react-bootstrap/Nav';
 import React from 'react';
+import { getRepoList, addToRepoList } from '../util/GithubRepositoryUtil';
 
-class MusicMarkdownNavbar extends React.Component {
-  constructor(props) {
-    super(props);
-    this.repoList = [{
-      'owner': 'music-markdown',
-      'repo': 'almost-in-time',
-      'path': '/'
-    }];
-  }
+const darkLightThemeFlag = 'dark';
+const REPO_LIST_KEY = 'repoList';
 
-  render() {
-    return this.createNavbar();
+const MusicMarkdownNavbar = () => {
+  if (!localStorage.getItem(REPO_LIST_KEY)) {
+    addToRepoList('music-markdown', 'almost-in-time', '/');
   }
+  return createNavbar();
+};
 
-  createNavbar() {
-    const darkLightThemeFlag = 'dark';
-    return (
-      <Navbar bg={darkLightThemeFlag} expand="lg" variant={darkLightThemeFlag} key="test">
-        <Navbar.Brand href="/">Music Markdown</Navbar.Brand>
-        <Navbar.Toggle aria-controls="basic-navbar-nav" />
-        <Navbar.Collapse>
-          <Nav className="mr-auto">
-            <NavDropdown title="Music Repositories">
-              {/* TODO: Build edit repo functionality */}
-              <NavDropdown.Item><b>Edit Repositories</b></NavDropdown.Item>
-              <NavDropdown.Divider />
-              {this.getRepositoriesDropdownItems()}
-            </NavDropdown>
-            <Nav.Link href="#/sandbox">Sandbox (Beta)</Nav.Link>
-          </Nav>
-        </Navbar.Collapse>
-      </Navbar>
-    );
-  }
+function createNavbar() {
+  return (
+    <Navbar bg={darkLightThemeFlag} expand="lg" variant={darkLightThemeFlag} key="test">
+      <Navbar.Brand href="/">Music Markdown</Navbar.Brand>
+      <Navbar.Toggle aria-controls="basic-navbar-nav" />
+      <Navbar.Collapse>
+        <Nav className="mr-auto">
+          <NavDropdown title="Music Repositories">
+            {/* TODO: Build edit repo functionality */}
+            <NavDropdown.Item><b>Edit Repositories</b></NavDropdown.Item>
+            <NavDropdown.Divider />
+            {getRepositoriesDropdownItems()}
+          </NavDropdown>
+          <Nav.Link href="#/sandbox">Sandbox (Beta)</Nav.Link>
+        </Nav>
+      </Navbar.Collapse>
+    </Navbar>
+  );
+}
 
-  /**
-   * For all added repositories, add it to the dropdown list
-   */
-  // TODO: List first x items, then put in dropdown item to expand full list
-  getRepositoriesDropdownItems() {
-    const repoDropdownItems = [];
-    this.repoList.forEach(function(repo) {
+/**
+ * For all added repositories, add it to the dropdown list
+ */
+// TODO: List first x items, then put in dropdown item to expand full list
+function getRepositoriesDropdownItems() {
```

Comment:
So, what makes this more logically a react component, and not `createNavbar()`? I think I'm not fully versed in what be composed as a React component.

Reply:
In the React documentation, they decompose larger components by moving a block of code into a sub-component. It has the advantage that you can re-use the component elsewhere (if that ever comes up), because it's entirely self-contained. 

The current way leaks state. It requires that the outer component by matched with <NavDropdown>.